airflow/include/tasks/extract/utils/html_utils.py

```diff
@@ -23,28 +24,72 @@ def is_valid_url(url):
     return bool(parsed.netloc) and bool(parsed.scheme)
 
 
-def exclude_path(url, exclude_docs=None):
+def fetch_page_content(url: str) -> str:
+    try:
+        response = requests.get(url)
+        response.raise_for_status()  # Raise an HTTPError for bad responses
+        return response.content
+    except requests.RequestException:
+        logger.info("Error fetching content for %s: %s", url, url)
+        return ""
+
+
+def exclude_url(url: str, exclude_literal: list[str]) -> bool:
+    url_path = urlparse(url).path
+    return any(literal in url_path for literal in exclude_literal)
+
+
+def remove_tags(text_content: str) -> str | None:
     """
-    Check if the URL path contains any of the specified strings to be excluded.
+    Clean the HTML content by removing script and style tags, collapsing whitespaces, and extracting text.
 
-    param url (str): The URL to check.
-    param exclude_docs (list): List of strings to exclude from the URL path.
+    param text_content (str): The HTML content to be cleaned.
     """
-    if exclude_docs is None:
-        exclude_docs = []
-    url_path = urlparse(url).path
-    return any(docs in url_path for docs in exclude_docs)
+    soup = BeautifulSoup(text_content, "html.parser").find("body")
 
+    if soup is None:
+        return
+    # Remove script and style tags
+    for script_or_style in soup(["script", "style"]):
+        script_or_style.extract()
+
+    # Get text and handle whitespaces
+    text = " ".join(soup.stripped_strings)
+
+    return text
 
-def get_all_links(url, exclude_docs=None):
+
+def truncate_tokens(text: str, encoding_name: str = "gpt-3.5-turbo", max_length: int = 8192) -> str | None:
+    """
+    Truncates a text string based on the maximum number of tokens.
+
+    param string (str): The input text string to be truncated.
+    param encoding_name (str): The name of the encoding model.
+    param max_length (int): The maximum number of tokens allowed. Default is 8192.
+    """
+    import tiktoken
```

Comment:
May I know why we import tiktoken here?

Reply:
just dag parsing passing time will have some effect but should be significant, let me know i'm ok to move it on top too

Reply:
I'm not sure how heavy this lib is, but I'm ok with either way. just want to confirm it. thanks!

Reply:
Yes lets test it @pankajastro